File: android/src/com/dontflymoney/baseactivity/Message.java
package com.dontflymoney.baseactivity;

import android.app.AlertDialog;
import android.app.ProgressDialog;
import android.content.DialogInterface;
import android.content.DialogInterface.OnClickListener;

import com.dontflymoney.view.R;

public class Message
{
	SmartActivity activity;
	
	Message(SmartActivity activity)
	{
		this.activity = activity;
	}
	
	
	public void alertError(Object message)
	{
		alertError(message.toString());
	}
	
	public void alertError(int resourceId, Exception e)
	{
		alertError(activity.getString(resourceId)+ ": " + e.getLocalizedMessage());
	}
	
	private void alertError(String message)
	{
		alertError(message, new OnClickListener(){
			@Override
			public void onClick(DialogInterface dialog, int which) {
				dialog.cancel();
			}
    	});
	}
	
	private void alertError(String message, OnClickListener clickListener)
	{
		new AlertDialog.Builder(activity)
			.setTitle(R.string.error_title)
			.setMessage(message)
			.setPositiveButton(R.string.ok_button, clickListener)
    		.show();
	}
	
	public void alertRetryLicense()
	{
		alertError(activity.getString(R.string.license_retry),
			new OnClickListener(){
				@Override
				public void onClick(DialogInterface dialog, int which) {
					dialog.cancel();
					activity.refresh();
				}
	    	}
		);
	}
	
	


	public ProgressDialog getWaitDialog()
	{
		ProgressDialog progress = new ProgressDialog(activity);
		progress.setTitle(activity.getString(R.string.wait_title));
		progress.setMessage(activity.getString(R.string.wait_text));
		
		return progress;
	}
	
	public ProgressDialog showWaitDialog()
	{
		ProgressDialog progress = getWaitDialog();
		progress.show();
		
		return progress;
	}
	
	
	
}

File: android/DFM/src/main/kotlin/com/darakeon/activities/base/Message.kt
package com.dontflymoney.baseactivity

import android.app.AlertDialog
import android.app.ProgressDialog
import android.content.DialogInterface
import android.content.DialogInterface.OnClickListener
import com.dontflymoney.activityObjects.SmartStatic

import com.dontflymoney.view.R

class Message<T : SmartStatic> internal constructor(private val activity: SmartActivity<T>) {


    fun alertYesNo(message: String, answer: IYesNoDialogAnswer) {
        alertError(message, R.string.ok_button, true, OnClickListener { dialog, which ->
            when (which) {
                DialogInterface.BUTTON_POSITIVE -> answer.YesAction()

                DialogInterface.BUTTON_NEGATIVE -> answer.NoAction()
            }

            dialog.cancel()
        })
    }


    fun alertError(message: Any) {
        alertError(message.toString())
    }

    fun alertError(resMessage: Int, e: Exception) {
        alertError(activity.getString(resMessage) + ": " + e.message)
    }

    private fun alertError(message: String, resOkButton: Int = R.string.ok_button, hasCancelButton: Boolean = false, clickListener: OnClickListener = OnClickListener { dialog, which -> dialog.cancel() }) {
        val builder = AlertDialog.Builder(activity)
                .setTitle(R.string.error_title)
                .setMessage(message)

        if (resOkButton != 0)
            builder.setPositiveButton(resOkButton, clickListener)

        if (hasCancelButton)
            builder.setNegativeButton(R.string.cancel_button, clickListener)

        builder.show()
    }

    fun alertRetryLicense() {
        val message = activity.getString(R.string.license_retry)

        alertError(message, R.string.try_again, true, OnClickListener { dialog, which ->
            dialog.cancel()
            activity.refresh()
        })
    }


    internal val waitDialog: ProgressDialog
        get() {
            val progress = ProgressDialog(activity)
            progress.setTitle(activity.getString(R.string.wait_title))
            progress.setMessage(activity.getString(R.string.wait_text))

            return progress
        }

    fun showWaitDialog(): ProgressDialog {
        val progress = waitDialog
        progress.show()

        return progress
    }


}
